src/charm.py

```diff
@@ -122,18 +142,25 @@ def _is_config_valid(self) -> bool:
             errors.append("'saml_sync_groups' cannot be specified without a 'saml_target_url'")
 
         if self.config.get("s3_enabled"):
-            [
-                errors.append(f"'s3_enabled' requires '{s3_config}'")
-                for s3_config in REQUIRED_S3_SETTINGS
-                if not self.config[s3_config]
-            ]
+            errors.extend(
+                [
```

Comment:
Would it be worth making this a generator? It should just require changing the brackets from `[` to `(`, that way the list isn't created and then absorbed into `errors` and instead the elements are put straight into the list. Something like the following should work:

```Python
errors.extend(f"'s3_enabled' requires '{s3_config}'" for s3_config in REQUIRED_S3_SETTINGS if not self.config[s3_config)
```

Reply:
Done